util/sqlexec/restricted_sql_executor.go

```diff
@@ -94,6 +94,10 @@ type RecordSet interface {
 	// Close closes the underlying iterator, call Next after Close will
 	// restart the iteration.
 	Close() error
+
+	// LogPartialSlow logs the slow log for partial result.
+	// it will be used in server-side cursor or multi-queries.
+	LogPartialSlow()
```

Comment:
The method name is too specific.
Something like `OnFetchDone` would be better.

Reply:
I tried remove this method out of that interface and give a new name now~ PTAL if free